features/enlinkd/org.opennms.features.enlinkd.topogen/src/main/java/org/opennms/enlinkd/topogen/TopologyGenerator.java

```diff
@@ -0,0 +1,228 @@
+    } else {
+      throw new IllegalArgumentException("Don't know this protocol: " + this.protocol);
+    }
+  }
+
+  private static void assertMoreOrEqualsThan(String message, int expected, int actual) {
+    if (actual < expected) {
+      throw new IllegalArgumentException(message + String.format(" minimum expected=%s but found actual=%s", expected, actual));
+    }
+  }
+
+  public static class TopologyGeneratorBuilder {
+
+    private TopologyPersister persister;
+    private Integer amountNodes;
+    private Integer amountElements;
+    private Integer amountLinks;
+    private Integer amountSnmpInterfaces;
+    private Integer amountIpInterfaces;
+    private Topology topology;
+    private Protocol protocol;
+    private Boolean deleteExistingTolology;
+
+    private TopologyGeneratorBuilder() {
+    }
+
+    public TopologyGeneratorBuilder persister(TopologyPersister persister) {
+      this.persister = persister;
+      return this;
+    }
+
+    public TopologyGeneratorBuilder amountNodes(Integer amountNodes) {
+      this.amountNodes = amountNodes;
+      return this;
+    }
+
+    public TopologyGeneratorBuilder amountElements(Integer amountElements) {
+      this.amountElements = amountElements;
+      return this;
+    }
+
+    public TopologyGeneratorBuilder amountLinks(Integer amountLinks) {
+      this.amountLinks = amountLinks;
+      return this;
+    }
+
+    public TopologyGeneratorBuilder amountSnmpInterfaces(Integer amountSnmpInterfaces) {
+      this.amountSnmpInterfaces = amountSnmpInterfaces;
+      return this;
+    }
+
+    public TopologyGeneratorBuilder amountIpInterfaces(Integer amountIpInterfaces) {
+      this.amountIpInterfaces = amountIpInterfaces;
+      return this;
+    }
+
+    public TopologyGeneratorBuilder topology(Topology topology) {
+      this.topology = topology;
+      return this;
+    }
+
+    public TopologyGeneratorBuilder protocol(Protocol protocol) {
+      this.protocol = protocol;
+      return this;
+    }
+
+    public TopologyGeneratorBuilder deleteExistingTolology(
+        Boolean deleteExistingTolology) {
+      this.deleteExistingTolology = deleteExistingTolology;
+      return this;
+    }
+
+    public TopologyGenerator build() {
+      return new TopologyGenerator(persister, amountNodes, amountElements, amountLinks,
+          amountSnmpInterfaces, amountIpInterfaces, topology, protocol, deleteExistingTolology);
+    }
+
+    public String toString() {
+      return "TopologyGenerator.TopologyGeneratorBuilder(persister=" + this.persister
+          + ", amountNodes=" + this.amountNodes + ", amountElements=" + this.amountElements
+          + ", amountLinks=" + this.amountLinks + ", amountSnmpInterfaces="
+          + this.amountSnmpInterfaces + ", amountIpInterfaces=" + this.amountIpInterfaces
+          + ", topology=" + this.topology + ", protocol=" + this.protocol
+          + ", deleteExistingTolology=" + this.deleteExistingTolology + ")";
+    }
```

Comment:
Please use the Google Guava `MoreObjects.ToStringHelper` instead

Reply:
fixed